pkg/discovery/monitoring/kubelet/kubelet_monitor.go

```diff
@@ -152,17 +152,19 @@ func (m *KubeletMonitor) scrapeKubelet(node *api.Node) {
 		glog.Warningf("Failed to get kubelet thresholds for %s, %v.", node.Name, err)
 	}
 
-	metricFamilies, err := kc.GetCPUThrottlingMetrics(ip, node.Name)
-	if err != nil {
-		glog.Warningf("Failed to read kubelet cadvisor metrics for %s, %v.", node.Name, err)
-	}
-	if _, found := metricFamilies[kubeclient.ContainerCPUThrottledTotal]; !found {
-		glog.V(3).Infof("No throttling metrics found for node %s.", node.Name)
-	}
-
 	// TODO Use time stamp attached to the discovered CPUStats/MemoryStats of node and pod from kubelet to be more precise
 	currentMilliSec := time.Now().UnixNano() / int64(time.Millisecond)
-	m.generateThrottlingMetrics(metricFamilies, currentMilliSec)
+	if utilfeature.DefaultFeatureGate.Enabled(features.ThrottlingMetrics) {
```

Comment:
Should we disable by default for private preview?

Reply:
The feature flag that we implemented, we did decide to keep only disabling as an option (as far as I remember for brevity) for kubeturbo, ie. be able to disable a given metrics collection only if need be. I have left this as is in this PR. If we want to enhance the feature flag, and have features disabled by default I can extend the feature flag and update this one to false in a separate PR. Does that make sense?
@enlinxu you can also comment on this please.

Reply:
@evat-pm can chime in here, but I think she wants throttling metrics by default which is in the acceptance criteria too.

Reply:
@irfanurrehman We may want to negate the logic for feature gate to better serve the private preview purpose, we can talk more later.
@chlam4 In terms of the default here, I want to make sure the market part is private preview only. So I need to confirm with Shravan if a separate feature flag will be used for the analysis side.

Reply:
@enlinxu completely agree interms of the feature gate feature. I will create a separate task (outside the current story) to update the feature gates this way, while also disabling the throttling metrics collection by default.

Reply:
Thanks @irfanurrehman . Agreed that let's handle it separately. I have talked to Shravan and he's looking into adding the feature flag on server side and will get back on Monday. If he's able to add feature flag in market, then we can keep this enabled by default!